benchmarks/python/core.py

```diff
@@ -185,7 +186,8 @@ def _get_kernel_time(
                 if hasattr(event, "self_device_time_total")
                 else event.self_cuda_time_total
             )
-        assert has_cuda_event, "No CUDA events found"
+        # This assert happens when running thunder backward for rope benchmark.
+        # assert has_cuda_event, "No CUDA events found"
```

Comment:
@Priya2698 what does this assert failure mean? strangely after commenting it out, the benchmark still seem to run fine and reporting numbers that looks about right.

Reply:
This should be fixed by PR #3600. It will be merged in a couple hours.

Reply:
You are the best!

Reply:
Nope. I'm still hitting this. :cry: 
after commenting it out, I'm still hitting some division by zero thing as well...

Reply:
Yeah the division by zero is because the measured timing is zero.
Let me try out this PR.

Reply:
I cannot reproduce this -- which set of parameters (executor, rope_variation) is this happening? All benchmarks run for me:
![Screenshot 2024-12-19 at 4 39 31 PM](https://github.com/user-attachments/assets/072abcd7-5494-46b7-8cc2-7ab225572313)

Reply:
Looks like it indeed worked for you.

I'm seeing my local log to still report benchmark number even when the assert happens, but the number looks a bit strange sometimes with that. Meanwhile your number looks perfectly normal. :confused: 

```
------------------------------------------------------------------------------------------------------------------------------- benchmark: 5 tests ------------------------------------------------------------------------------------------------------------------------------
Name (time in us)                                                                                       Min                   Max                  Mean             StdDev                Median                IQR            Outliers         OPS            Rounds  Iterations
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
test_rope_variations_fwd_benchmark[executor='thunder'-rope_variation='hf_mistral_nemo_rope']       370.2880 (1.0)        443.9610 (1.01)       425.6754 (1.0)      31.2098 (inf)        437.1610 (1.0)      24.6095 (inf)           1;1  2,349.2079 (1.0)           5           1
test_rope_variations_fwd_benchmark[executor='thunder'-rope_variation='hf_qwen2_rope']              437.4830 (1.18)       437.4830 (1.0)        437.4830 (1.03)      0.0000 (1.0)        437.4830 (1.00)      0.0000 (1.0)           0;0  2,285.8031 (0.97)          1           1
test_rope_variations_fwd_benchmark[executor='thunder'-rope_variation='hf_phi3_rope']               806.7370 (2.18)       806.7370 (1.84)       806.7370 (1.90)      0.0000 (1.0)        806.7370 (1.85)      0.0000 (1.0)           0;0  1,239.5613 (0.53)          1           1
test_rope_variations_fwd_benchmark[executor='thunder'-rope_variation='llama_2_7b_hf_rope']         831.6490 (2.25)     1,005.5420 (2.30)       933.9474 (2.19)     55.0864 (inf)        930.6845 (2.13)     70.6170 (inf)           3;0  1,070.7241 (0.46)         10           1
test_rope_variations_fwd_benchmark[executor='thunder'-rope_variation='llama_3_8B_rope']          1,702.8520 (4.60)     1,743.1810 (3.98)     1,716.9607 (4.03)     18.5003 (inf)      1,710.9050 (3.91)     25.9475 (inf)           1;0    582.4245 (0.25)          4           1
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

Legend:
  Outliers: 1 Standard Deviation from Mean; 1.5 IQR (InterQuartile Range) from 1st Quartile and 3rd Quartile.
  OPS: Operations Per Second, computed as 1 / Mean
======================================== short test summary info ========================================
FAILED test_rope.py::test_rope_variations_fwd_benchmark[executor='thunder'-rope_variation='llama_3_8B_rope'] - AssertionError: No CUDA events found
FAILED test_rope.py::test_rope_variations_fwd_benchmark[executor='thunder'-rope_variation='hf_qwen2_rope'] - AssertionError: No CUDA events found
FAILED test_rope.py::test_rope_variations_fwd_benchmark[executor='thunder'-rope_variation='hf_phi3_rope'] - AssertionError: No CUDA events found
FAILED test_rope.py::test_rope_variations_fwd_benchmark[executor='thunder'-rope_variation='hf_mistral_nemo_rope'] - AssertionError: No CUDA events found
```

Reply:
everything seems to work for me in today's container. I guess there's probably some stale stuff in last night's container, which doesn't have your fix before my rebuild